File: local_examples/cmds_servermgmt/seredis/CmdsServerMgmtExample.cs
// EXAMPLE: cmds_servermgmt
using System;
using System.Linq;
using StackExchange.Redis;

public class CmdsServerMgmt
{
    public void Run()
    {
        // FLUSHALL and INFO are server-admin commands, so connect with
        // allowAdmin=true and issue them through the IServer object.
        var muxer = ConnectionMultiplexer.Connect("localhost:6379,allowAdmin=true");
        var server = muxer.GetServer("localhost:6379");

        // STEP_START flushall
        server.FlushAllDatabases();

        var res2 = server.Keys(pattern: "*").ToArray();
        Console.WriteLine(res2.Length); // >>> 0
        // STEP_END

        // STEP_START info
        var res3 = server.Info();
        Console.WriteLine(res3.Length > 0); // >>> True
        // STEP_END
    }
}

File: local_examples/cmds_servermgmt/lettuce-reactive/CmdsServerMgmtExample.java
// EXAMPLE: cmds_servermgmt
package io.redis.examples.reactive;

import io.lettuce.core.RedisClient;
import io.lettuce.core.api.reactive.RedisReactiveCommands;
import io.lettuce.core.api.StatefulRedisConnection;
// REMOVE_START
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;
// REMOVE_END

import reactor.core.publisher.Mono;

public class CmdsServerMgmtExample {

    // REMOVE_START
    @Test
    // REMOVE_END
    public void run() {
        RedisClient redisClient = RedisClient.create("redis://localhost:6379");

        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
            RedisReactiveCommands<String, String> reactiveCommands = connection.reactive();

            // STEP_START flushall
            Mono<Void> flushallExample = reactiveCommands.flushall()
                    .doOnNext(res1 -> {
                        System.out.println(res1); // >>> OK
                        // REMOVE_START
                        assertThat(res1).isEqualTo("OK");
                        // REMOVE_END
                    })
                    .then(reactiveCommands.keys("*").collectList())
                    .doOnNext(res2 -> {
                        System.out.println(res2); // >>> []
                        // REMOVE_START
                        assertThat(res2).isEmpty();
                        // REMOVE_END
                    })
                    .then();
            // STEP_END

            flushallExample.block();

            // STEP_START info
            Mono<Void> infoExample = reactiveCommands.info()
                    .doOnNext(res3 -> {
                        System.out.println(res3);
                        // >>> # Server
                        // >>> redis_version:7.4.0
                        // >>> ...
                        // REMOVE_START
                        assertThat(res3).contains("redis_version");
                        // REMOVE_END
                    })
                    .then();
            // STEP_END

            infoExample.block();
        } finally {
            redisClient.shutdown();
        }
    }
}
